### ISSUE TITLE
Some DNS not correctly resolved

### ISSUE THREAD

**Issue by jbenoit2011** [mongodb/mongo-php-driver#387]
### Description

I tried to connect to a mongo cluster with following domain name `internal-mongodb-elb-internal-1111111111.eu-west-1.elb.amazonaws.com` and it failed. Domain name was resolved to `internal-mongodb-elb-internal-1111111111.eu-west-1.elb.amazonaws.internal-mongodb-elb-internal-1111111111.eu-west-1.elb.amazonaws.com:2:27017`.

Then, I created a much shorter alias `mongo.foobar.com` and it worked.

So I wonder maybe there is bug when parsing long domain name.

### Environment

PHP 7.0.9 in a docker container.

```
mongodb
mongodb support => enabled
mongodb version => 1.1.8
mongodb stability => stable
libmongoc version => 1.3.5
libbson version => 1.3.5
mongodb.debug => no value => no value
```

### Test Script

From the CLI

Not working :(
```
<?php
php -r "var_dump((new \MongoDB\Driver\Manager('mongodb://internal-mongodb-elb-internal-1111111111.eu-west-1.elb.amazonaws.com'))->executeCommand('admin', new \MongoDB\Driver\Command(['ping' => 1]))->toArray()[0]);"
```

Working :)
```
<?php
php -r "var_dump((new \MongoDB\Driver\Manager('mongodb://mongo.foobar.com'))->executeCommand('admin', new \MongoDB\Driver\Command(['ping' => 1]))->toArray()[0]);"
```

Happens in both HTTP and CLI context

### Expected and Actual Behavior

```
PHP Fatal error:  Uncaught MongoDB\Driver\Exception\ConnectionTimeoutException: No suitable servers found (`serverselectiontryonce` set): [Failed connecting to 'internal-mongodb-elb-internal-1111111111.eu-west-1.elb.amazonaws.internal-mongodb-elb-internal-1111111111.eu-west-1.elb.amazonaws.com:2:27017': php_network_getaddresses: getaddrinfo failed: Name or service not known] in Command line code:1
Stack trace:
#0 Command line code(1): MongoDB\Driver\Manager->executeCommand('admin', Object(MongoDB\Driver\Command))
#1 {main}
  thrown in Command line code on line 1
```

### Debug Log

```
[2016-09-06T10:55:23+00:00]     PHONGO: DEBUG   > Connection string: 'mongodb://internal-mongodb-elb-internal-1111111111.eu-west-1.elb.amazonaws.com'
[2016-09-06T10:55:23+00:00]     PHONGO: TRACE   > ENTRY: php_phongo_make_mongo_client():1924
[2016-09-06T10:55:23+00:00]     PHONGO: DEBUG   > Creating Manager, phongo-1.1.8[stable] - mongoc-1.3.5(bundled), libbson-1.3.5(bundled), php-7.0.9-1~dotdeb+8.1
[2016-09-06T10:55:23+00:00]     mongoc: TRACE   > ENTRY: mongoc_topology_description_init():51
[2016-09-06T10:55:23+00:00]     mongoc: TRACE   >  EXIT: mongoc_topology_description_init():68
[2016-09-06T10:55:23+00:00]     mongoc: TRACE   > ENTRY: mongoc_server_description_init():87
[2016-09-06T10:55:23+00:00]     mongoc: TRACE   >  EXIT: mongoc_server_description_init():123
[2016-09-06T10:55:23+00:00]    cluster: TRACE   > ENTRY: mongoc_cluster_init():1597
[2016-09-06T10:55:23+00:00]    cluster: TRACE   >  EXIT: mongoc_cluster_init():1620
[2016-09-06T10:55:23+00:00]     client: DEBUG   > Using custom stream initiator.
[2016-09-06T10:55:23+00:00]     PHONGO: TRACE   >  EXIT: php_phongo_make_mongo_client():2025
[2016-09-06T10:55:23+00:00]     cursor: TRACE   > ENTRY: _mongoc_cursor_new():85
[2016-09-06T10:55:23+00:00]     cursor: TRACE   >  EXIT: _mongoc_cursor_new():235
[2016-09-06T10:55:23+00:00]     cursor: TRACE   > ENTRY: mongoc_cursor_next():964
[2016-09-06T10:55:23+00:00]     cursor: TRACE   > TRACE: mongoc_cursor_next():969 cursor_id(0)
[2016-09-06T10:55:23+00:00]     cursor: TRACE   > ENTRY: _mongoc_cursor_next():1025
[2016-09-06T10:55:23+00:00]     cursor: TRACE   > ENTRY: _mongoc_cursor_initial_query():346
[2016-09-06T10:55:23+00:00]     cursor: TRACE   > ENTRY: _mongoc_cursor_fetch_stream():305
[2016-09-06T10:55:23+00:00]    cluster: TRACE   > ENTRY: _mongoc_cluster_stream_for_optype():1685
[2016-09-06T10:55:23+00:00]     PHONGO: TRACE   > ENTRY: phongo_stream_initiator():1143
[2016-09-06T10:55:23+00:00]     PHONGO: DEBUG   > Connecting to 'internal-mongodb-elb-internal-1111111111.eu-west-1.elb.amazonaws.internal-mongodb-elb-internal-1111111111.eu-west-1.elb.amazonaws.com:2:27017[mongodb://internal-mongodb-elb-internal-1111111111.eu-west-1.elb.amazonaws.com]'
[2016-09-06T10:55:23+00:00]     PHONGO: TRACE   >  EXIT: phongo_stream_initiator():1198
[2016-09-06T10:55:23+00:00]     mongoc: TRACE   > ENTRY: mongoc_server_description_handle_ismaster():383
[2016-09-06T10:55:23+00:00]     mongoc: TRACE   >  EXIT: mongoc_server_description_handle_ismaster():389
[2016-09-06T10:55:23+00:00]     mongoc: TRACE   > TRACE: mongoc_topology_description_handle_ismaster():1400 No transition entry to 4 for 0
[2016-09-06T10:55:23+00:00]     mongoc: TRACE   > ENTRY: mongoc_topology_description_select():493
[2016-09-06T10:55:23+00:00]     mongoc: TRACE   >  EXIT: mongoc_topology_description_select():507
[2016-09-06T10:55:23+00:00]    cluster: TRACE   >  EXIT: _mongoc_cluster_stream_for_optype():1697
[2016-09-06T10:55:23+00:00]     cursor: TRACE   >  EXIT: _mongoc_cursor_fetch_stream():322
[2016-09-06T10:55:23+00:00]     cursor: TRACE   >  GOTO: _mongoc_cursor_initial_query():353 done
[2016-09-06T10:55:23+00:00]     cursor: TRACE   >  EXIT: _mongoc_cursor_initial_query():382
[2016-09-06T10:55:23+00:00]     cursor: TRACE   >  EXIT: _mongoc_cursor_next():1079
[2016-09-06T10:55:23+00:00]     cursor: TRACE   >  EXIT: mongoc_cursor_next():1000
[2016-09-06T10:55:23+00:00]     cursor: TRACE   > ENTRY: mongoc_cursor_error():923
[2016-09-06T10:55:23+00:00]     cursor: TRACE   > ENTRY: _mongoc_cursor_error():941
[2016-09-06T10:55:23+00:00]     cursor: TRACE   >  EXIT: _mongoc_cursor_error():951
[2016-09-06T10:55:23+00:00]     cursor: TRACE   >  EXIT: mongoc_cursor_error():933
[2016-09-06T10:55:23+00:00]     cursor: TRACE   > ENTRY: mongoc_cursor_destroy():242
[2016-09-06T10:55:23+00:00]     cursor: TRACE   > ENTRY: _mongoc_cursor_destroy():259
[2016-09-06T10:55:23+00:00]     cursor: TRACE   >  EXIT: _mongoc_cursor_destroy():296
[2016-09-06T10:55:23+00:00]     cursor: TRACE   >  EXIT: mongoc_cursor_destroy():252
[2016-09-06T10:55:23+00:00]     mongoc: TRACE   > ENTRY: mongoc_topology_description_destroy():89
[2016-09-06T10:55:23+00:00]     mongoc: TRACE   > ENTRY: mongoc_server_description_destroy():145
[2016-09-06T10:55:23+00:00]     mongoc: TRACE   >  EXIT: mongoc_server_description_destroy():151
[2016-09-06T10:55:23+00:00]     mongoc: TRACE   >  EXIT: mongoc_topology_description_destroy():103
[2016-09-06T10:55:23+00:00]    cluster: TRACE   > ENTRY: mongoc_cluster_destroy():1643
[2016-09-06T10:55:23+00:00]    cluster: TRACE   >  EXIT: mongoc_cluster_destroy():1653
```

Maybe the problem comes from `libmongo` and not this PHP extension. I stumbled upon this closed ticket in mongo C driver JIRA  : https://jira.mongodb.org/browse/CDRIVER-1129?jql=text%20~%20%22uri%20parsing%22

**Comment by jbenoit2011** [comment]
It seems the issue come from libmongoc. So I close this ticket for the moment. Feel free to reopen it.

**Comment by jmikola** [comment]
Thanks for the detailed report. [CDRIVER-1129](https://jira.mongodb.org/browse/CDRIVER-1129) will add the error during URI parsing and should be included in libmongoc 1.5.0, which will be used by our next minor version (1.2.0).
